### PR TITLE
memory calculation with deltas (proof of concept)

### PR DESCRIPTION
**Pull request by Sovetnikov**

### CODE REVIEW

**Comment by fabianp** [comment]
Thanks! Note that there are some issues with unicode and python2.7 (see travis failure)

File: memory_profiler.py [hunk 1]
```diff
@@ -520,12 +520,18 @@ def add(self, code, toplevel_code=None):
...
             self.add(subcode, toplevel_code=toplevel_code)
 
-    def trace(self, code, lineno):
+    def trace(self, code, lineno, prev_lineno):
         memory = _get_memory(-1, self.backend, include_children=self.include_children,
                              filename=code.co_filename)
-        # if there is already a measurement for that line get the max
-        previous_memory = self[code].get(lineno, 0)
-        self[code][lineno] = max(memory, previous_memory)
+        prev_value = self[code].get(lineno, None)
```
Comment (sauravbiswasiupr) [inline memory_profiler.py]: this should by default give None

**Comment by Juanlu001** [comment]
Hit by #122 today. @Sovetnikov are you still working on this? @fabianp do you want me to take over?

**Comment by fabianp** [comment]
@Juanlu001 yes please!

**Comment by Juanlu001** [comment]
Closing in favor of #166. Thanks @Sovetnikov!

**Pull request closed without merging**